crmsh/utils.py

```diff
@@ -2512,34 +2473,141 @@ def check_file_content_included(source_file, target_file):
-    if remote_addr:
-        # check on remote
-        prompt_msg = "Check whether {} is active on {}".format(service, remote_addr)
-        rc, _, _ = run_cmd_on_remote(cmd, remote_addr, prompt_msg)
-    else:
-        # check on local
-        rc, _ = get_stdout(cmd)
-    return rc == 0
+    def _do_action(self, action_type):
+        """
+        Actual do actions to manage service
+        """
+        if action_type not in self.ACTION_LIST:
+            raise ValueError("status_type should be {}".format('/'.join(self.ACTION_LIST)))
+
+        cmd = "systemctl {} {}".format(action_type, self.service_name)
+        if self.remote_addr:
+            prompt_msg = "Run \"{}\" on {}".format(cmd, self.remote_addr)
+            rc, output, err = run_cmd_on_remote(cmd, self.remote_addr, prompt_msg)
+        else:
+            rc, output, err = get_stdout_stderr(cmd)
+        if rc != 0 and err:
+            raise ValueError("Run \"{}\" error: {}".format(cmd, err))
+        return rc == 0, output
+
+    def is_available(self):
+        return self.service_name in self._do_action("list-unit-files")[1]
+
+    def is_enabled(self):
+        return self._do_action("is-enabled")[0]
+
+    def is_active(self):
+        return self._do_action("is-active")[0]
+
+    def start(self):
+        self._do_action("start")
+
+    def stop(self):
+        self._do_action("stop")
+
+    def enable(self):
+        self._do_action("enable")
+
+    def disable(self):
+        self._do_action("disable")
+
+    @classmethod
+    def service_is_available(cls, name, remote_addr=None):
+        """
+        Check whether service is available
+        """
+        inst = cls(name, remote_addr)
+        return inst.is_available()
+
+    @classmethod
+    def service_is_enabled(cls, name, remote_addr=None):
+        """
+        Check whether service is enabled
+        """
+        inst = cls(name, remote_addr)
+        return inst.is_enabled()
+
+    @classmethod
+    def service_is_active(cls, name, remote_addr=None):
+        """
+        Check whether service is active
+        """
+        inst = cls(name, remote_addr)
+        return inst.is_active()
+
+    @classmethod
+    def start_service(cls, name, remote_addr=None):
+        """
+        Start service
+        """
+        inst = cls(name, remote_addr)
+        inst.start()
+
+    @classmethod
+    def stop_service(cls, name, remote_addr=None):
+        """
+        Stop service
+        """
+        inst = cls(name, remote_addr)
+        inst.stop()
+
+    @classmethod
+    def enable_service(cls, name, remote_addr=None):
+        """
+        Enable service
+        """
+        inst = cls(name, remote_addr)
+        inst.enable()
+
+    @classmethod
+    def disable_service(cls, name, remote_addr=None):
+        """
+        Disable service
+        """
+        inst = cls(name, remote_addr)
+        inst.disable()
+
+    @classmethod
+    def enable_start_service(cls, name, remote_addr=None):
```

Comment:
This name looks pretty strange.
I would only have `start_service` and add a parameter named `enabled=False`, or something like that. And if this parameters comes, just run the `enable` method.
What do you think?

Reply:
Ok, changed as `enable` parameter in `start_service` function, and `disable` parameter in `stop_service` function